libdnf/dnf-keyring.cpp

```diff
@@ -252,9 +278,27 @@ dnf_keyring_check_untrusted_file(rpmKeyring keyring,
         goto out;
     }
 
-    /* we don't want to abort on missing keys */
     ts = rpmtsCreate();
-    rpmtsSetVSFlags(ts, _RPMVSF_NOSIGNATURES);
+
+    if (rpmtsSetKeyring(ts, keyring) < 0) {
+        g_set_error_literal(error, DNF_ERROR, DNF_ERROR_INTERNAL_ERROR, "failed to set keyring");
+        goto out;
+    }
+    rpmtsSetVfyLevel(ts, RPMSIG_SIGNATURE_TYPE | RPMSIG_DIGEST_TYPE);
+    rpmlogSetCallback(rpmcliverifysignatures_log_handler_cb, &rpm_error);
```

Comment:
This makes libdnf officially not thread-safe, but there isn’t a better option I know of other than fork/exec of `/usr/bin/rpmkeys`.  Perhaps we should document that all uses of libdnf/librpm/etc need to be wrapped in a mutex.

Reply:
Yes, but we already use this approach in `dnf_rpmts_find_package` so it's not new.